qa/comparitor.py

```diff
@@ -0,0 +1,180 @@
+import csv
+import requests
+from datetime import datetime, timedelta
+
+from dictdiffer import diff
+
+from qa.metrics import METRICS
+
+class MetricDiff(object):
+    state = None
+    county = None
+    fips = None
+    date = None
+    metric_name = None
+
+    snapshot1 = None
+    snapshot2 = None
+
+    value1 = None
+    value2 = None
+
+    threshold = None
+    difference = None
+
+    def __init__(self, state, county, fips, date, metric_name, snapshot1, snapshot2, value1, value2, difference, threshold):
+        self.state  = state
+        self.county = county
+        self.fips = fips
+        self.date = date
+        self.metric_name = metric_name
+        self.snapshot1 = snapshot1
+        self.snapshot2 = snapshot2
+        self.value1 = value1
+        self.value2 = value2
+        self.threshold = threshold
+        self.difference = difference
+
+
+class Comparitor(object): 
+    snapshot1 = None
+    snapshot2 = None
+
+    state = None
+    fips = None
+
+    api1 = None
+    api2 = None
+
+    _date_to_data = {}
+    _results = {}
+    
+    def _get_state_data(self, snapshot1, snapshot2, state_abbrev, intervention): 
+        # counties_url_1 = f'https://data.covidactnow.org/snapshot/{snapshot1}/us/counties.{intervention}.timeseries.json'
+        # counties_url_2= f'https://data.covidactnow.org/snapshot/{snapshot2}/us/counties.{intervention}.timeseries.json'
+
+        states_url_1 = f'https://data.covidactnow.org/snapshot/{snapshot1}/us/states/{state_abbrev}.{intervention}.timeseries.json'
+        states_url_2 = f'https://data.covidactnow.org/snapshot/{snapshot2}/us/states/{state_abbrev}.{intervention}.timeseries.json'
+
+        # counties_json_1 = requests.get(counties_url_1).json()
+        # counties_json_2 = requests.get(counties_url_2).json()
+
+        states_json_1 = requests.get(states_url_1).json()
+        states_json_2 = requests.get(states_url_2).json()
+
+        return states_json_1, states_json_2
+
+    def __init__(self, snapshot1, snapshot2, state, intervention, fips = None):
+        self.snapshot1 = snapshot1
+        self.snapshot2 = snapshot2
+        self.state = state
+        self.fips = fips
+        self.intervention = intervention
+        # TODO (sgoldblatt) swap out here if it's counties
+        self.api1, self.api2 = self._get_state_data(snapshot1, snapshot2, state, intervention)
+        self._generate_date_dictionary()
```

Comment:
Thoughts on separating out the loading of the state data from the comparitor? 

Would be great to pass in the loaded data dictionary to separate concerns a bit more. Also makes it easier to test this class without having to hit the actual url.